docs/rules-engine/README.md

```diff
@@ -0,0 +1,169 @@
+* an operator
+* [optional] a right operand (FACT, RUNTIME_FACT or LITERAL)
+
+The operator is a method that will evaluate whether a condition is met. It can take up to two operands.
+
+While the Otter already provides a set of operators ready for use, it is also possible to use your own operator that will
+fit your needs. You can check on the [custom operator documentation](./how-to-use/custom-operator.md) as well as the 
+[ruleset generation documentation](./how-to-use/industrialize-ruleset-generation.md) if you want to generate metadata for your own operators.
+
+The exhaustive list of exposed operators is available in the ``rules.operators.metadata.json``.
+Note that this file is automatically generated via our extractors and matches the list of operators exposed
+in the [module](https://github.com/AmadeusITGroup/otter/blob/main/packages/%40o3r/rules-engine/src/engine/operator/operators/index.ts), this is why it is only available in the npm artifact.
+
+> [!NOTE]
+> For the operators comparing a text variable to a pattern (such as `matchesPattern`, `oneMatches`, and `allMatch`),
+> we support the ES RegExp `/^myRegExp.*$/i` (containing the pattern and optional flags) or just the RegExp content `^myregexp.*$`.
+>
+> The special characters used in the pattern should contain a double backslash (`\\`).
+> For example, to check if a string contains a `\t`, the pattern would need to include `\\t`.
+>
+> Also, to avoid the wrong detection of an ES RegExp instead of RegExp content, a content beginning with a slash `/` character
+> (such as a path `/path/to/file`) should be preceded by a double backslash `\\` (for example `\\/path/to/file`)
+
+### Action
+
+``ActionBlock`` describes the action to execute. Each ``Action`` is identified
+by a unique id to identify the process to trigger and an eventual payload.
+
+The Otter framework provides a few build-in actions linked to Otter specific features:
+* ``UPDATE_CONFIG`` modifies the configuration of a configurable component
+(provided by [@o3r/configuration](https://www.npmjs.com/package/@o3r/configuration))
+* ``UPDATE_ASSET`` overrides the path of an asset which uses the ``o3rDynamicContent`` pipe 
+(provided by [@o3r/dynamic-content](https://www.npmjs.com/package/@o3r/dynamic-content))
+* ``UPDATE_LOCALISATION`` substitutes the translation of a key with the one from another key.
+([@o3r/localization](https://www.npmjs.com/package/@o3r/localization))
+* ``UPDATE_PLACEHOLDER`` injects html code in a specific placeholder
+([@o3r/components](https://www.npmjs.com/package/@o3r/components))
+
+You can provide your own custom action following [this guide](./how-to-use/custom-action.md).
+
+> [!NOTE]
+> The actions handled by the rules engine need to be [registered](./how-to-use/README.md#step-2-register-the-action-handlers-in-the-runner) in your application to be recognized by the service.
+> This ensures you only import packages you actually needs.
+
+### Fact
```

Comment:
Maybe we should have the Fact paragraph before the Rule one because we use several time the "fact" concept in the Rule section